app/services/invoices/progressive_billing_service.rb

```diff
@@ -17,9 +17,14 @@ def call
 
         invoice.fees_amount_cents = invoice.fees.sum(:amount_cents)
         invoice.sub_total_excluding_taxes_amount_cents = invoice.fees_amount_cents
-        Credits::AppliedCouponsService.call(invoice:) if invoice.fees_amount_cents&.positive?
+
+        if invoice.fees_amount_cents.positive?
+          Credits::ProgressiveBillingService.call(invoice:)
```

Comment:
🤔 when/why would we not end up crediting the previously invoiced usage?

Reply:
Good catch, it should always g through the Credits services :+1: